Recursive/Josephus.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @function Josephus
+ * @description recursive implementation of the Josephus function.
+ * @param {Integer[]} collection - The integer array.
+ * @param {Integer} - The step size.
+ * @return {Integer} - The last integer in the list.
+ * @see [JosephusProblem](https://en.wikipedia.org/wiki/Josephus_problem)
+ * @example [1,2,3,4,5,6,7] with step 3 = 4
+ */
+
+const josephus = (collection, step) => {
+  // return null for invalid steps that are less than or equal to 0
+  if (step <= 0 || collection.length === 0) {
+    return null
+  }
+  if (collection.length === 1) {
+    return collection[0]
+  } else {
+    step = (step - 1) % collection.length
+    collection.splice(step, 1)
```

Comment:
This is $O(n)$ so you get $O(n^2)$ runtime, which is very naive. Wikipedia shows formulae that achieve $O(n)$ or $O(k\log{n})$ runtime, both of which are substantially better. And even besides those, I can pretty easily come up with $O(n \log{n})$ algorithms based on decremental data structures which support this "splice" operation in logarithmic time (such as an initially balanced k-nary tree augmented with counts).

This algorithm also doesn't really benefit from recursion. It would be clearer as a simple `while` loop:

```javascript
while (collection.length > 1) { /* kill someone */ }
return collection[0]
```